instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex

```diff
@@ -125,14 +121,36 @@ defmodule OpentelemetryEcto do
 
     OpenTelemetry.Span.end_span(s)
 
-    if parent_context != :undefined do
-      OpenTelemetry.Ctx.detach(parent_context)
-    end
+    if ctx?(parent_context), do: OpenTelemetry.Ctx.detach(parent_context)
   end
 
   defp format_error(%{__exception__: true} = exception) do
     Exception.message(exception)
   end
 
   defp format_error(_), do: ""
+
+  defp ctx?(:undefined), do: false
+  defp ctx?(_), do: true
+
+  defp maybe_add(map, _, nil), do: map
+  defp maybe_add(map, key, value) when is_map(map), do: Map.put(map, key, value)
+
+  defp maybe_append_source(name, nil), do: name
+  defp maybe_append_source(name, source), do: "#{name}:#{source}"
+
+  defp maybe_add_peer_name(map, peer_name) do
+    if ip?(peer_name), do: map, else: maybe_add(map, "net.peer.name", peer_name)
```

Comment:
I'm not sure that this logic is correct. Looking at [opentelemetry-specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#netpeername)
I see these two statements:

> Sometimes host name is only available to instrumentation as a string which may contain DNS name or IP address. net.peer.name SHOULD be set to the available known hostname (e.g., "127.0.0.1" if connecting to an URL https://127.0.0.1.com/foo).

> If only IP address is available via net.sock.peer.addr, net.peer.name SHOULD NOT be set.

I think I have this wrong. I think this means we SHOULD always set it in our case regardless of if it is only an ip address.